pkg/didcomm/common/service/service.go

```diff
@@ -21,6 +21,13 @@ type Handler interface {
 	HandleOutbound(msg *DIDCommMsg, myDID, theirDID string) error
 }
 
+// Headerable provides common methods for the message
+type Headerable interface {
+	MsgID() string
```

Comment:
I don't actually see these methods being used?

Reply:
Nothing is being used for now. My point is to generalize all messages in the framework. `MsgID` will be used for incrementing a counter (`sender_order`), but not in this PR

Reply:
An accessor should not have side effects. You should use a different method for a counter.

Reply:
I think we shouldn't add an interface until they have a clear usage. I would move this change to another PR, where it is used, so we can evaluate it together.

E.g., You need a counting mechanism. This seems somewhat separate from what has been defined.